test/unit/postprocessing/conftest.py

```diff
@@ -0,0 +1,173 @@
+import numpy as np
+import pandas as pd
+import pytest
+from fairlearn.postprocessing._threshold_operation import ThresholdOperation
+from fairlearn.postprocessing._constants import SCORE_KEY, LABEL_KEY, SENSITIVE_FEATURE_KEY
+
+from test.unit.constants import MULTIPLE_SENSITIVE_FEATURE_COMPRESSION_SKIP_REASON
+from test.unit.input_convertors import ensure_list_1d, ensure_ndarray, ensure_ndarray_2d, \
+    ensure_dataframe, ensure_series, _map_into_single_column
+
+
+X_ex = np.stack(([0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19],
+                 [5, 4, 3, 2, 7, 8, 3, 4, 4, 3, 6,  5,  7,  5,  2,  1,  9,  4,  8,  0],
+                 [9, 2, 4, 2, 9, 3, 1, 8, 1, 1, 2,  9,  6,  7,  4,  2,  56, 1,  2,  34]), -1)
+
+sensitive_feature_names_ex1 = ["A", "B", "C"]
+sensitive_features_ex1 = np.array([x for x in 'AAAAAAA' 'BBBBBBB' 'CCCCCC']).reshape(-1, 1)
+
+sensitive_feature_names_ex2 = ["x", "Y"]
+sensitive_features_ex2 = np.array([x for x in 'xxxYYYY' 'xYYYYYx' 'YYYYYY']).reshape(-1, 1)
+
+labels_ex = np.array([int(x) for x in '0110100' '0010111' '000111'])
+degenerate_labels_ex = np.array([int(x) for x in '0000000' '0000000' '000000'])
+scores_ex = np.array([int(x) for x in '0011233' '0001111' '011112'])
+
+# combine earlier examples for a new scenario with multiple sensitive features
+sensitive_features_ex3 = np.hstack((sensitive_features_ex1, sensitive_features_ex2))
+sensitive_feature_names_ex3 = ["A,x", "A,Y", "B,x", "B,Y", "C,Y"]
+
+candidate_X_transforms = [ensure_ndarray, ensure_dataframe]
+candidate_Y_transforms = [ensure_list_1d, ensure_ndarray, ensure_series, ensure_dataframe]
+candidate_A_transforms = [ensure_list_1d, ensure_ndarray, ensure_ndarray_2d, ensure_series,
+                          ensure_dataframe]
+
+LabelAndPrediction = namedtuple('LabelAndPrediction', 'label prediction')
+
+_data = namedtuple('_data', 'example_name feature_names sensitive_features X y scores')
+
+
+@pytest.fixture(params=[
+    _data("example 1",
+          sensitive_feature_names_ex1,
+          sensitive_features_ex1,
+          X_ex,
+          labels_ex,
+          scores_ex),
+    _data("example 2",
+          sensitive_feature_names_ex2,
+          sensitive_features_ex2,
+          X_ex,
+          labels_ex,
+          scores_ex),
+    _data("example 3",
+          sensitive_feature_names_ex3,
+          sensitive_features_ex3,
+          X_ex,
+          labels_ex,
+          scores_ex)])
+def data(request):
+    return request.param
+
+
+@pytest.fixture(params=candidate_A_transforms)
+def data_sf(data, request):
+    sensitive_feature_transform = request.param
+    # Skip combinations where the multi-column sensitive features would have to be compressed
+    # into a one-dimensional data structure.
+    if (data.sensitive_features == sensitive_features_ex3).all() and \
+            sensitive_feature_transform in [ensure_list_1d, ensure_series]:
+        pytest.skip(MULTIPLE_SENSITIVE_FEATURE_COMPRESSION_SKIP_REASON)
```

Comment:
Is there no way we can prevent the invalid combinations being generated? Seeing skipped tests makes me think we're either got unfixed bugs (and hence are suppressing the test until the fix is in) or unimplemented features (similarly).

Reply:
It's not super straightforward but I'll see what I can do

Reply:
Is this acceptable to you?
```
1488 passed, 192 deselected in 285.46 seconds